tests/benchmark/test_worst_compute.py

```diff
@@ -2317,3 +2317,95 @@ def test_worst_return_revert(
         post={},
         tx=tx,
     )
+
+
+@pytest.mark.valid_from("Osaka")
+def test_worst_clz_same_input(
+    state_test: StateTestFiller, blockchain_test: BlockchainTestFiller, pre: Alloc, fork: Fork
+):
+    """Test running a block with as many CLZ with same input as possible."""
+    env = Environment()
+    attack_gas_limit = env.gas_limit
+    tx_gas_limit = fork.transaction_gas_limit_cap()
+
+    magic_value = 248  # CLZ(248) = 248
+
+    calldata = Op.PUSH1(magic_value)
+    attack_block = Op.CLZ
+    code = code_loop_precompile_call(calldata, attack_block, fork)
+    code_address = pre.deploy_contract(code=code)
+
+    tx = Transaction(
+        to=code_address,
+        gas_limit=tx_gas_limit if tx_gas_limit else attack_gas_limit,
+        sender=pre.fund_eoa(),
+    )
+
+    if (tx_gas_limit is None) or (tx_gas_limit > attack_gas_limit):
+        state_test(
+            env=env,
+            pre=pre,
+            post={},
+            tx=tx,
+        )
+    else:
+        tx_count = attack_gas_limit // tx_gas_limit
+        txs = [tx for _ in range(tx_count)]
+
+        blockchain_test(
+            genesis_environment=Environment(),
+            pre=pre,
+            post={},
+            blocks=[Block(txs=txs)],
+        )
+
+
+@pytest.mark.valid_from("Osaka")
+def test_worst_clz_diff_input(
+    state_test: StateTestFiller,
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+):
+    """Test running a block with as many CLZ with different input as possible."""
+    env = Environment()
+    attack_gas_limit = env.gas_limit
+    tx_gas_limit = fork.transaction_gas_limit_cap()
+    max_code_size = fork.max_code_size()
+
+    code_prefix = Op.JUMPDEST
+    code_suffix = Op.PUSH0 + Op.JUMP
+
+    iteration = (max_code_size - len(code_prefix) - len(code_suffix)) // 2
+
+    code_seq = Bytecode()
+
+    for i in range(2 * iteration):
+        value = i if i % 2 else 2**256 - 1 - i
+        code_seq += Op.CLZ(value) + Op.POP
+
+    code_address = pre.deploy_contract(code=code_prefix + code_seq + code_suffix)
```

Comment:
This will deploy a contract too large. Could you add the sanity check?

```suggestion
    code_address = pre.deploy_contract(code=code_prefix + code_seq + code_suffix)
    if len(attack_code) > max_contract_size:
        raise ValueError(
            f"Code size {len(attack_code)} exceeds maximum code size {max_contract_size}"
        )
```

Reply:
I might go with the following approach for consistency now, since the remaining benchmarks use it as well. 

However, we could add this refactoring for all the assertions later in this [issue](https://github.com/ethereum/execution-spec-tests/issues/1704)

```py
 assert len(attack_code) <= max_code_size
```